src/walk-manifest.js

```diff
@@ -19,6 +19,15 @@ const fsSanitize = function(filepath) {
     .join(path.sep);
 };
 
+const urlBasename = function(uri) {
+  const parsed = url.parse(uri);
+  const pathname = parsed.pathname || parsed.path.replace(parsed.query || '', '');
+  const query = (parsed.query || '').split(/\\\\|\\|\//).join('!');
```

Comment:
replace path like characters in the query with `!` to maintain some semblance of what it was.